core/consensus/grandpa/impl/voting_round_impl.cpp

```diff
@@ -0,0 +1,663 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "consensus/grandpa/impl/voting_round_impl.hpp"
+
+#include <boost/range/adaptors.hpp>
+#include <boost/range/algorithm/find_if.hpp>
+#include <boost/range/numeric.hpp>
+#include "common/visitor.hpp"
+#include "primitives/justification.hpp"
+
+namespace kagome::consensus::grandpa {
+
+  static auto convertToPrimaryPropose = [](const auto &vote) {
+    return PrimaryPropose(vote.block_number, vote.block_hash);
+  };
+
+  static auto convertToPrevote = [](const auto &vote) {
+    return Prevote(vote.block_number, vote.block_hash);
+  };
+
+  static auto convertToBlockInfo = [](const auto &vote) {
```

Comment:
Er... Don't quite get why are their lambdas instead of plain functions.

Reply:
We can use `auto`s instead of tempates. I love it, don't think it is too critical